pool.go

```diff
@@ -166,6 +178,70 @@ func (p *pool) refreshPool() {
 	}
 }
 
+func (p *pool) checkPool() {
+	for {
+		select {
+		case msg := <-p.mirrorSamples:
+			// see if it is to a main-tier node - if so find appropriate test node to test against.
+			p.lk.RLock()
+			if p.th.NodeTier(msg.node) != tieredhashing.TierMain {
+				p.lk.RUnlock()
```

Comment:
Why do we need to see if this is a `TierMain ` node ? Will an unknown node request make it to the loop ? Oh if we don't have enough main nodes for the mainset requests ?

Reply:
yeah, if we're already using unknown nodes because we don't have a good main pool then we don't need to also be doing mirror-based sampling.